Mergin/plugin.py

```diff
@@ -162,61 +170,87 @@ def add_action(
+                return True
+        return False
 
     def on_config_changed(self):
         """Called when plugin config (connection settings) were changed."""
         self.create_manager()
         self.enable_toolbar_actions()
 
-    def connect_provider_root_item(self):
-        """Set the connection for Mergin config changes."""
-        if self.data_item_provider.root_item is not None:
-            # first try to disconnect the root item signal
-            self.disconnect_provider_root_item()
-            self.data_item_provider.root_item.config_changed.connect(self.on_config_changed)
-            self.data_item_provider.root_item.local_project_removed.connect(self.on_config_changed)
-
-    def disconnect_provider_root_item(self):
-        try:
-            self.data_item_provider.root_item.config_changed.disconnect(self.on_config_changed)
-            self.data_item_provider.root_item.local_project_removed.disconnect(self.on_config_changed)
-        except (TypeError, AttributeError):
-            pass
-
     def enable_toolbar_actions(self, enable=None):
         """Check current project and set Mergin Toolbar icons enabled accordingly."""
         if enable is None:
             enable = mergin_project_local_path() is not None
-        if self.data_item_provider.root_item.mc is None:
+        if self.manager is None:
             enable = False
         for action in self.toolbar.actions():
             if action.text() in self.actions_always_on:
                 action.setEnabled(True)
             elif action.text() == "Create Mergin Project":
-                can_create = self.data_item_provider.root_item.mc is not None and self.manager is not None
-                action.setEnabled(can_create)
+                action.setEnabled(self.mc is not None and self.manager is not None)
             else:
                 action.setEnabled(enable)
 
+    def show_browser_panel(self):
+        """Check if QGIS Browser panel is open. If not, ask and eventually make it visible to users."""
+        browser = [w for w in self.iface.mainWindow().findChildren(QDockWidget) if w.objectName() == "Browser"][0]
+        q = "QGIS Browser panel is currently off. The panel is used for Mergin projects management.\n"
+        q += "Would you like to open it and see your Mergin projects?"
+        res = QMessageBox.question(None, "Mergin - QGIS Browser Panel", q)
+        if not browser.isVisible() and res == QMessageBox.Yes:
+            self.iface.addDockWidget(Qt.LeftDockWidgetArea, browser)
+
     def configure(self):
         """Open plugin configuration dialog."""
-        if self.data_item_provider.root_item is None:
-            QMessageBox.warning(
-                None,
-                "Mergin Plugin",
-                "Mergin plugin first use requires QGIS restart.",
-            )
-            return
-        self.data_item_provider.root_item.configure()
+        self.show_browser_panel()
```

Comment:
:thinking:  not sure if we need this... and especially to do it before opening the config dialog... I would at least move that further down when the config dialog was accepted...

Reply:
Moved down.